ArcGISRuntimeSDKQt_QMLSamples/Analysis/FindFeaturesUtilityNetwork/FindFeaturesUtilityNetwork.qml

```diff
@@ -0,0 +1,358 @@
+// [WriteFile Name=FindFeaturesUtilityNetwork, Category=Analysis]
+// [Legal]
+// Copyright 2019 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+import QtQuick 2.6
+import QtQuick.Layouts 1.12
+import QtQuick.Controls 2.12
+import Esri.ArcGISRuntime 100.7
+
+Rectangle {
+    id: rootRectangle
+    clip: true
+    width: 800
+    height: 600
+
+    readonly property url featureLayerUrl: "https://sampleserver7.arcgisonline.com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer"
+    property UtilityElement element
+    property UtilityNetworkSource utilityNetworkSource
+    property list<UtilityTerminal> startingLocations
+    property list<UtilityTerminal> barriers
+    property list<UtilityTerminal> terminals
+    property UtilityTerminal terminal
+    property ArcGISFeature identifiedFeature
+    property Point clickPoint
+    property var deviceObjIds: []
+    property var lineObjIds: []
+    property UtilityElementTraceResult myTraceResult
+
+    MapView {
+        id: mapView
+        anchors.fill: parent
+        
+        Map {
+            BasemapStreetsNightVector {}
+            
+            ViewpointExtent {
+                Envelope {
+                    xMin: -9813547.35557238
+                    yMin: 5129980.36635111
+                    xMax: -9813185.0602376
+                    yMax: 5130215.41254146
+                    spatialReference: SpatialReference.createWebMercator();
+                }
+            }
+
+            // Add the layer with electric distribution lines.
+            FeatureLayer {
+                id: lineLayer
+
+                ServiceFeatureTable {
+                    url: featureLayerUrl + "/115"
+                }
+
+                renderer: SimpleRenderer {
+                    symbol: featureLineSymbol
+                }
+
+                onSelectFeaturesStatusChanged: checkSelectionStatus();
+            }
+
+            // Add the layer with electric devices.
+            FeatureLayer {
+                id: deviceLayer
+
+                ServiceFeatureTable {
+                    url: featureLayerUrl + "/100"
+                }
+
+                onSelectFeaturesStatusChanged: checkSelectionStatus();
+            }
+        }
+
+        GraphicsOverlay {
+            id: unGraphicsOverlay
+        }
+
+        onMouseClicked: {
+            let screenX = mouse.x;
```

Comment:
the default should probably be const in most cases. I'd suggest making that your default, and using let when you need to reassign or have block scoped variables